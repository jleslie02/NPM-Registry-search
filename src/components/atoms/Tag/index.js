/** @jsx jsx */
// eslint-disable-next-line no-unused-vars
import React from 'react';
import { jsx, css } from '@emotion/core';
import PropTypes from 'prop-types';

const propTypes = {
  label: PropTypes.string
};
const defaultProps = {
  label: ''
};

export const StatelessTag = props => {
  const { label, theme } = props;

  const tag = css(
    (() => ({
      padding: '6px 14px',
      borderRadius: '20px',
      fontSize: '14px',
      border: '1px solid #c0c4d2',
      color: 'rgba(80, 80, 80, 0.87)',
      cursor: 'pointer',
      backgroundColor: '#fff',
      ...theme.mixins.flexCenter()
    }))()
  );

  return (
    <div css={tag} data-gm="tag">
      {' '}
      {label}{' '}
    </div>
  );
};

StatelessTag.propTypes = propTypes;

StatelessTag.propTypes = propTypes;
StatelessTag.defaultProps = defaultProps;
export default StatelessTag;
