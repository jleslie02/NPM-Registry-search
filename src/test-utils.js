import React from "react";
import { render } from "react-testing-library";
import makeAtomicTheme from "./theme";

export const themedRender = (MyComponent, props) => {
  return render(<MyComponent theme={makeAtomicTheme()} {...props} />);
};

export const themedComponent = (MyComponent, props) => {
  return <MyComponent theme={makeAtomicTheme()} {...props} />;
};
