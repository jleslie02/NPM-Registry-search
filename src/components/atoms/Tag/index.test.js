// Static component with conditional rendering so snapshot test should suffice
import renderer from "react-test-renderer";
import Tag from "./index";
import { themedComponent } from "../../../test-utils";

it("renders correctly with full props", () => {
  const props = {
    label: "A label",
  };
  const tree = renderer.create(themedComponent(Tag, props)).toJSON();
  expect(tree).toMatchSnapshot();
});
