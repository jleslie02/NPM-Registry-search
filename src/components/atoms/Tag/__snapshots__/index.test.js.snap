// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with full props 1`] = `
<div
  className="css-89ujr-StatelessTag"
  data-testid="tag"
>
   
  A label
   
</div>
`;
